src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,78 @@
-import React from 'react';
-import { Loader } from '../Loader';
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../Loader/Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as removeTodoAction } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const dispatch = useAppDispatch();
+  const todo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    if (todo?.userId) {
+      getUser(todo.userId)
+        .then(userFS => setUser(userFS))
+        .catch(() => {
+          throw new Error('User not found on server!');
+        })
+        .finally(() => setIsLoading(false));
+    }
+  }, [todo?.userId]);
+
+  const removeCurrentTodo = () => {
+    dispatch(removeTodoAction.removeTodo());
+    setUser(null);
+  };
+
   return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+    <>
+      {todo && (
+        <div className="modal is-active" data-cy="modal">
+          <div className="modal-background" />
+
+          {isLoading && <Loader />}
```

Comment:
Loader is showing on top of the modal menu. Better use ternary operator here

Reply:
Good point!